### ISSUE TITLE
Precondition violation with arrangement + CGAL 5.2

### ISSUE THREAD

**Issue by colormotor** [scikit-geometry/scikit-geometry#62]
Hi, 
I am trying to create an arrangement on Ubuntu, with libcgal-dev 5.2 installed.
``` python
import skgeom
from skgeom import Point2, Segment2, Polygon
import numpy as np
import matplotlib.pyplot as plt

w = 100
P_1 = [[0,0.0], [w, 0.1], [w,w], [0,w], [0,0.0]] # <----
# P_1 = [[0,0.0], [w, 0.0], [w,w], [0,w], [0,0.0]] # This works
P_2 = [[x+w*0.5, y+w*0.5] for x,y in P_1]

arr = skgeom.arrangement.Arrangement()

for P in [P_1, P_2]:
    for p, q in zip(P, P[1:]):
        try:
          arr.insert(skgeom.Segment2(skgeom.Point2(*p),
                                   skgeom.Point2(*q)))
        except Exception as e:
          print(e)

def _point_to_np(p):
    return np.array([float(p.x()),
                 float(p.y())])
    
def face_vertices(face):
    if face.is_unbounded():
        return []
    P = []
    i = face.outer_ccb
    first = next(i)
    halfedge = next(i)
    while first != halfedge:
        P.append(_point_to_np(halfedge.source().point()))
        halfedge = next(i)
    P.append(_point_to_np(halfedge.source().point()))
    return P

plt.figure()
skgeom.draw.draw_polygon(skgeom.Polygon(P_1[:-1]), facecolor='none')
skgeom.draw.draw_polygon(skgeom.Polygon(P_2[:-1]), facecolor='none')
for face in arr.faces:
    P = face_vertices(face)
    if P:
        skgeom.draw.draw_polygon(skgeom.Polygon(P), facecolor=np.ones(3)*np.random.uniform(0, 0.5))
plt.axis('equal')
plt.show()
```

This breaks with
```
CGAL ERROR: precondition violation!
Expr: (m_traits.compare_y_at_x_2_object()(p, cv) == EQUAL) && compare_xy(cv.left(), p) == SMALLER && compare_xy(cv.right(), p) == LARGER
File: /usr/include/CGAL/Arr_segment_traits_2.h
Line: 610
```
which is caused by the 0.1 in the definition of `P_1` above.

The same code works perfectly on OSX with CGAL version 5.0.3.

I'm new to CGAL, but perhaps this has to do with stricter checks in the new version?

**Comment by wolfv** [comment]
It looks like you're inserting the same point twice into the polygon. is that right? if you remove the duplicate point, does it work?

**Comment by ab1nash** [comment]
@colormotor thanks for pointing out this issue. I bypassed this issue by commenting this line out and then reinstalling from the repository.

**Comment by colormotor** [comment]
@wolfv I am trying to create a square-like shape so two segments share a point. Strange thing this works perfectly with an older version of CGAL, but not the latest.

**Comment by colormotor** [comment]
I wander if this has something to do with the switch to inexact kernel in commit e9ca3aba622057c820eaac565be5f6a0dc869953 ? I tried to recompile with exact but I get a number of errors.

**Comment by colormotor** [comment]
@wolfv so I did manage to compile with CGAL 5.2+ and `typedef CGAL::Exact_predicates_exact_constructions_kernel   Kernel;` and this solves the problem. To compile I had to specify the kernel in the call to `CGAL::create_interior_straight_skeleton_2` for the definition of  `create_interior_straight_skeleton` in `skeleton.cpp`.  	

For the moment, since I don't need this, I also commented the definition of `offset_polygons`, since the exact kernel seems to conflict with `CGAL::create_offset_polygons_2`. 

Happy to create a pull request if you think this is useful.
